internal/manifest/raw_ostree.go

```diff
@@ -61,7 +63,24 @@ func (p *RawOSTreeImage) serialize() osbuild.Pipeline {
 
 	inputName := "root-tree"
 	copyOptions, copyDevices, copyMounts := osbuild.GenCopyFSTreeOptions(inputName, p.treePipeline.Name(), p.Filename, pt)
-	copyInputs := osbuild.NewPipelineTreeInputs(inputName, p.treePipeline.Name())
+
+	commitChecksum := p.treePipeline.commit.Checksum
+
+	treeInputs := osbuild.NewPipelineTreeInputs(inputName, p.treePipeline.Name())
+	copyInputs := *treeInputs
+
+	bootFiles := p.platform.GetBootFiles()
+	if len(bootFiles) > 0 {
+		for _, paths := range bootFiles {
+			copyOptions.Paths = append(copyOptions.Paths, osbuild.CopyStagePath{
+				From: fmt.Sprintf("input://ostree-tree/%s/%s", commitChecksum, paths[0]),
+				To:   fmt.Sprintf("mount://root/%s", paths[1]),
+			})
+		}
+
+		copyInputs["ostree-tree"] = osbuild.NewOSTreeCheckoutInput("org.osbuild.source", commitChecksum)
```

Comment:
The origin is IMO wrong here, since the input origin is a pipeline, not source. We should really add dedicated function to generate the pipeline origin input, such as it is done for the files input (https://github.com/osbuild/osbuild-composer/blob/a56c70d5053b1c2e5293d68dc3665eefe694829f/internal/osbuild/files_input.go#L136) and not pass the origin as an argument.

Reply:
We discussed this off-channel and I didn't realize that the copy stage is used in a bit unexpected way to copy files from an ostree commit which is taken as an input for the stage. The reason for this is that the copy stage can't copy files within the tree and using `Stage.MountOSTree()` won't work on pipeline origin input. So while not ideal, this approach is valid.